src/main/java/com/github/hokkaydo/eplbot/module/addcommand/CustomCommandManager.java

```diff
@@ -0,0 +1,127 @@
+    public final Long guildId;
+
+    class CommandItem {
+        String content;
+        String authorId;
+
+        public CommandItem(String content, String authorId) {
+            this.content = content;
+            this.authorId = authorId;
+        }
+
+        @Override
+        public String toString() {
+            return  '\'' + content + '\'' +
+                    " by " + authorId;
+        }
+    }
+
+    public CustomCommandManager(Long guildId) {
+        this.guildId = guildId;
+        loadCommands();
+    }
+
+    public void loadCommands(){
+        try(FileReader stream = new FileReader(commandsPath.toFile())) {
+            int c;
+            StringBuilder commandName = new StringBuilder();
+            StringBuilder commandContent = new StringBuilder();
+            StringBuilder authorId = new StringBuilder();
+            int index = 0;
+            while((c = stream.read()) != -1) {
+                if(c == ';') {
+                    index++;
+                    continue;
+                }
+                if(c == '\n') {
+                    commands.put(commandName.toString(), new CommandItem(commandContent.toString(), authorId.toString()));
+                    commandName = new StringBuilder();
+                    commandContent = new StringBuilder();
+                    authorId = new StringBuilder();
+                    index = 0;
+                    continue;
+                }
+                if (index == 0) commandName.append((char) c);
+                else if (index == 1) commandContent.append((char) c);
+                else if (index == 2) authorId.append((char) c);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        System.out.println(commands);
+    }
+
+    public List<Command> getCommands() {
+        List<Command> new_commands = new ArrayList<>();
+        commands.forEach((name, item) -> new_commands.add(new CustomCommand(name, item.content)));
+        System.out.println(new_commands);
+        return new_commands;
+    }
+
+    public void addCommand(Member author,  String commandName, String commandContent) {
+        //sanitize commandName
+        if(commandName.contains(";")) 
+            commandName = commandName.replace(";", "");
+
+        if(existsCommand(commandName)) return;
+        commands.put(commandName, new CommandItem(commandContent, author.getId()));
+        Main.getCommandManager().addCommands(author.getGuild(), List.of(new CustomCommand(commandName, commandContent)));
+        storeCommands();
+    }
+
+    public void removeCommand(Member author,  String commandName) {
+        //sanitize commandName
+        if(commandName.contains(";")) 
+            commandName = commandName.replace(";", "");
+
+        if(!existsCommand(commandName)) return;
+
+        CommandItem command = commands.get(commandName);
+
+        if(!command.authorId.equals(author.getId())) return;
```

Comment:
Why this check?
Admins will not be able to delete a command :thinking:

Reply:
I just tough it would be more logical that only the author of the command would be able to delete it ... but we can drop this mechanic and let anyone delete any CustomCommand